### PR TITLE
Move from setup.py to poetry.

### PR DESCRIPTION
**Pull request by KPLauritzen**

### CODE REVIEW

File: .github/workflows/code_coverage.yml [hunk 1]
```diff
@@ -17,22 +17,22 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
```
Comment (nybyhansen) [inline .github/workflows/code_coverage.yml:19]: consider using v3 :blush:

File: .github/workflows/code_coverage.yml [hunk 2]
```diff
@@ -17,22 +17,22 @@ jobs:
...
-    - name: Set up Python 3.9
+    - name: Set up Python 3.10
       uses: actions/setup-python@v2
```
Comment (nybyhansen) [inline .github/workflows/code_coverage.yml:21]: consider v4 :blush:

**Review by nybyhansen — DISMISSED** [review]
Looks good. I've just added some comments where I think we could also update the github actions used while we're updating the project a bit.

File: .github/workflows/python-publish.yml [hunk 3]
```diff
@@ -18,15 +18,9 @@ jobs:
...
-        python -m pip install --upgrade pip
-        pip install twine build
-    - name: Build and publish
-      env:
-        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-      run: |
-        python -m build
-        twine upload dist/*
+        python-version: '3.10'
```
Comment (pacanada) [inline .github/workflows/python-publish.yml:21]: Maybe double quotes to be consistent with the rest of python versions defined in the `yml`. I see it has changed to string in the pre-commit PR

**Review by pacanada — DISMISSED** [review]

File: .github/workflows/precommit-PR.yml [hunk 4]
```diff
@@ -17,28 +17,28 @@ jobs:
...
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
```
Comment (nybyhansen) [inline .github/workflows/precommit-PR.yml:26]:
```suggestion
      uses: snok/install-poetry@v1
      with:
        version: 1.1.10
        virtualenvs-create: false
```

File: .github/workflows/precommit-PR.yml [hunk 5]
```diff
@@ -17,28 +17,28 @@ jobs:
...
-      uses: actions/setup-python@v2
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
       with:
-        python-version: 3.8
+        python-version: "3.10"
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
```
Comment (nybyhansen) [inline .github/workflows/precommit-PR.yml:44]: Why are we using `pip` in this step? I think we should also pin the version of `pre-commit` here and make sure it corresponds to the one installed with `poetry`. Maybe its simpler to just stick with `poetry` here also? :blush:

File: .github/workflows/python-publish.yml [hunk 6]
```diff
@@ -14,19 +14,13 @@ jobs:
...
-        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-      run: |
-        python -m build
-        twine upload dist/*
+        python-version: "3.10"
+    - name: Build and publish to pypi
+      uses: JRubics/poetry-publish@v1.11
+      with:
+        repository_username: ${{ secrets.PYPI_USERNAME }}
```
Comment (nybyhansen) [inline .github/workflows/python-publish.yml:25]: I don't have permission to settings/secrets. Can you grant me access? ⭐

**Review by nybyhansen — APPROVED** [review]

File: .github/workflows/python-publish.yml (hunk 6 above)
Comment (KPLauritzen) [reply .github/workflows/python-publish.yml:25]: Done

File: .github/workflows/precommit-PR.yml (hunk 5 above)
Comment (KPLauritzen) [reply .github/workflows/precommit-PR.yml:44]: This is just to save time in running this. 
pre-commit does not need e.g. pandas to be installed to be able to run the checks. So I just wanted a quick install.

**Pull request merged**